src/client.cc

```diff
@@ -354,9 +354,18 @@ void Client::check_if_buffer_needs_reloading()
         return;
 
     const String& filename = buffer.name();
-    timespec ts = get_fs_timestamp(filename);
-    if (ts == InvalidTime or ts == buffer.fs_timestamp())
+    const timespec ts = get_fs_timestamp(filename);
+
+    if (ts == InvalidTime)
         return;
+    else
```

Comment:
No need for an else here, we already return in the if case.

Reply:
And check the filesize as well, so that we dont need to read the whole file if it obviously changed. (Checking fd.data.size() against the stored value should be good enough, mapping without actually reading should be fine).

Reply:
Good point.

I put `else` because it looked natural, since we're creating an anonymous scope regardless.

Reply:
You can use a if statement with initializer for that:
```
if (MappedFile fd{filename}; fd.st.st_size == status.size and hash_data(fd.data, fd.st.st_size) == status.hash)
    return;
```

Note that the use of `StringView{fd.data}` is wrong here, as it will look for a terminating 0-byte, which is not guaranteed and might be wrong (the file could contain zero bytes).